### PR TITLE
Support terraform_addons in audit recipe.

### PR DESCRIPTION
**Pull request by xingao267**

### CODE REVIEW

File: templates/tfengine/recipes/audit.hcl [hunk 1]
```diff
@@ -108,6 +108,12 @@ schema = {
         type = "string"
       }
     }
+    terraform_addons = {
+      description = <<EOF
+        Additional Terraform configuration for the cicd deployment.
```
Comment (pasha-gh) [inline templates/tfengine/recipes/audit.hcl:113]: for the `audit` deployment.

**Review by pasha-gh — APPROVED** [review]
minor documentation comment

File: templates/tfengine/recipes/audit.hcl (hunk 1 above)
Comment (xingao267) [reply templates/tfengine/recipes/audit.hcl]: Good catch! Fixed.

**Comment by thomasfung-hk** [comment]
What is the reason that we are restricting the provider versions to that range?

**Comment by xingao267** [comment]
> What is the reason that we are restricting the provider versions to that range?

No specific reason, just an example to demonstrate the custom provider constraints functionality. I added a comment.

**Comment by thomasfung-hk** [comment]
Looks like it failed the integration test while trying to create `ManagedZone`

**Review by thomasfung-hk — APPROVED** [review]

**Comment by xingao267** [comment]
> Looks like it failed the integration test while trying to create `ManagedZone`

That flake has been around for a while actually. Never got a chance to look into that more. If it happens more often, we can spend some bandwidth to fix it.

**Pull request merged**